src/CppUTestExt/MockSupport.cpp

```diff
@@ -310,20 +310,15 @@ void MockSupport::countCheck()
     UtestShell::getCurrent()->countCheck();
 }
 
-void MockSupport::checkExpectationsOfLastCall()
+void MockSupport::checkExpectations()
 {
+    for(MockNamedValueListNode *p = data_.begin(); p; p = p->next())
+        if(getMockSupport(p) && getMockSupport(p)->lastActualFunctionCall_)
```

Comment:
I'm afraid if lastActualFunctionCall is null then we lost chance to check other scope out of order.
for example:
```.cpp
    mock().strictOrder();
    mock("bla").strictOrder();
	mock().ignoreOtherCalls();

    MockExpectedCallsListForTest expectations;
    expectations.addFunction("foo1", 1)->callWasMade(2);
    expectations.addFunction("foo2", 2)->callWasMade(1);
    MockCallOrderFailure expectedFailure(mockFailureTest(), expectations);

    mock("bla").expectOneCall("foo1");
    mock("foo").expectOneCall("foo1");
    mock("foo").expectOneCall("foo2"); 

    mock("bla").actualCall("foo1");
    mock("foo").actualCall("foo2");
    mock("foo").actualCall("unexpected1");
    mock("foo").actualCall("foo1");
    mock("foo").actualCall("unexpected2");
```
I admit i'm fastidious about the out of order check :)

I have another proposed fix pull request, hopefully you guys can give some look and comments.